### DISCUSSION TITLE
Halting notebook execution when cells returns an error or throws unhandled exception

### DISCUSSION THREAD

**Discussion by andrerav** [dotnet/interactive#2402] [Q&A]
Hi!

I'm really loving .NET Interactive and I'm currently using it to set up some complex one-shot data migration pipelines that mixes C#, powershell and SQL.

I was wondering if there is a way to completely halt further notebook execution if a cell returns an error (i.e throws exception or a program returns an error code)? I've been looking through the docs but I can't seem to find a way to do this. 

Best regards,
Andreas

**Comment by jonsequitur** [comment]
We'll look into it. 

Are you using the VS Code extension? Also, are you interested in automating the execution of this kind of notebook from the command line?

**Reply by andrerav** [reply]
> We'll look into it.

Great, thank you, it would be a very practical feature :) 

> Are you using the VS Code extension?

Yes, I am using v1.0.3504060 (preview)

> Also, are you interested in automating the execution of this kind of notebook from the command line?

Yes! For example, it would be very nice to run this notebook in a continuous integration pipeline. If this hypothetically was possible today, then I would definitely set up a pipeline in our dev environment to run at least a few notebooks after merging PR's.

**Reply by jonsequitur** [reply]
You might find the automation features in [`dotnet-repl`](https://github.com/jonsequitur/dotnet-repl) interesting. It uses .NET Interactive and can run the same polyglot notebooks. We're using this to prototype potential automation features for the kernel.
